src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs

```diff
@@ -223,6 +261,74 @@ public class RestartComputerCommand : PSCmdlet, IDisposable
         [Alias("f")]
         public SwitchParameter Force { get; set; }
 
+        /// <summary>
+        /// Provide a comment for rebooting the computer.
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet)]
+        public string Comment { get; set; } = "";
+
+        /// <summary>
+        /// The major reason for rebooting the computer.
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet)]
+        [ValidateSet(
+            "Application",
+            "Hardware",
+            "OperatingSystem",
+            "Other",
+            "Power",
+            "Software",
+            "System")] 
+        public string MajorReason { get; set; }
+
+        /// <summary>
+        /// The minor reason for rebooting the computer.
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet)]
+        [ValidateSet(
+            "BlueScreen",
+            "Disk",
+            "Environment",
+            "Driver",
+            "HotFix",
+            "HotFixUninstall",
+            "Unresponsive",
+            "Installation",
+            "Maintenance",
+            "MMC",
+            "NetworkConnectivity",
+            "NetworkCard",
+            "Other",
+            "OtherDriver",
+            "PowerSupply",
+            "Processor",
+            "Reconfigure",
+            "SecurityIssue",
+            "SecurityPatch",
+            "SecurityPatchUninstallation",
+            "ServicePack",
+            "ServicePackUninstallation",
+            "TerminalServices",
+            "Unstable",
+            "Upgrade",
+            "WMI")]
+        public string MinorReason { get; set; }
```

Comment:
Per my other comment, you can skip ValidateSet entirely and you could opt to either have two enums (one for major and one for minor reason) or you could opt to roll it all into a single enum and parameter, depending on which is appropriate.

If multiple codes can be specified, tag the enum(s) with the `[Flags]` attribute to permit this.

Reply:
Seems we should review the list. How we can call the cmdlet if "BlueScreen"?

Reply:
I tried to go for parity with what the Win32ShutdownTracker supports, but it makes sense to remove some of the illogical ones. Do you have suggestions on what should be removed?

Reply:
I'd look options in Windows restart/shootdown dialog box.

Reply:
The options from the Server shutdown dialog include the following options. It seems a bit restrictive given what the API supports.

![image](https://user-images.githubusercontent.com/1432762/48989137-0280c680-f17d-11e8-87ee-d9f41b4f46a2.png)
![image](https://user-images.githubusercontent.com/1432762/48989153-10cee280-f17d-11e8-84b5-a4942297d8df.png)

Reply:
> It seems a bit restrictive given what the API supports.

because not all options make sense in interactive mode.

Reply:
Correct. So I would suggest not whittling it down to this list.

Reply:
Correct. So I would suggest not whittling it down to this list.